### DISCUSSION TITLE
How do I correctly apply a `span` in the middle of another tag?

### DISCUSSION THREAD

**Discussion by sylvesterroos** [mhanberg/temple#215] [Q&A] [answered]
I want to add some strikethrough text in my `p`, but I can't get it to work correctly. I've got the following snippet:
```elixir
p class: "text-lg" do
  "Oh, and maybe you've already noticed: this website has both a dark and a light theme, made possible by"
                                                                                                           
  span class: "line-through" do
    "my awesome CSS skills"
  end
                                                                                                           
  "TailwindCSS. Your eyes can thank me later 😌"
end
```

Which results in the following output:
![image](https://github.com/mhanberg/temple/assets/39904003/8c4d2291-17b1-4a28-a171-4d8fb748979d)

As you can see, the span is followed by a space, instead of being immediately followed by the rest of the `p`. Is this some Elixir stuff that I'm not aware of, or am I using Temple incorrectly? Thanks :)

**Comment by mhanberg** [comment]
It should just work I think, I have an example here: https://github.com/elixir-tools/elixir-tools.dev/blob/main/lib/elixir_tools/pages/home_page.ex

You can see it live at https://www.elixir-tools.dev

**Reply by mhanberg** [reply]
Wait, maybe I'm misunderstanding. Do you have a screen shot of the current behavior?

**Reply by sylvesterroos** [reply]
The current behaviour is my initial screenshot. After changing it to this:
```elixir
p class: "text-lg" do
  "Oh, and maybe you've already noticed: this website has both a dark and a light theme, made possible by"
                                                                                                           
  span class: "line-through", do: "my awesome CSS skills"
                                                                                                           
  "TailwindCSS. Your eyes can thank me later 😌"
end
```

I get the following output:
![image](https://github.com/mhanberg/temple/assets/39904003/f4e2e46c-a78d-411c-8685-b468a81395aa)

This is better, but not optimal. I would like to see the `TailwindCSS` directly after the `span` text, not with a space in between them

**Reply by mhanberg** [reply]
Oh sorry, I didn't realize that was a screenshot, just looked like text. 😅 

Can you confirm if you added the Temple compiler as noted here: https://hexdocs.pm/temple/getting-started.html#install and read the section on white space here: https://hexdocs.pm/temple/your-first-template.html#whitespace

**Reply by sylvesterroos** [reply]
1. Yup, the compiler is already added
2. I've never learned how HTML whitespace really works in plain HTML, let alone Temple 😅

Basically, I'm trying to recreate this in Temple: `<p>Hello <span style="text-decoration: line-through;">world</span>world</p>`

**Reply by sylvesterroos** [reply]
`mix temple.converter` gives me the same code as my own:
```
❯ cat /tmp/test.html && mix temple.convert /tmp/test.html
<p>Hello <span style="text-decoration: line-through;">world</span>world</p>
p do
  "Hello"

  span style: "text-decoration: line-through;" do
    "world"
  end

  "world"
end
```

Could you try to recreate this snippet on your end to rule out any potential wrong configurations on my end?

**Reply by mhanberg** [reply, marked as answer]
so in temple it just isn't possible (currently) for it to emit the code that you're looking for

a work around in this case is something like

```html
<p>
  Hello
  <span style="display: inline-flex;">
    <span style="text-decoration: line-through;">world</span>
    <span>world</span>
  </span>
</p>
```

in temple syntax this is

```elixir
p do
  "Hello"

  span style: "display: inline-flex" do
    span style: "text-decoration: line-through;", do: "world"
    span do: "world"
  end

  "world"
end
```

the main problem here I think is intuitively you want to interpolate the original span into the string, but that isn't supported currently.

**Reply by sylvesterroos** [reply]
Too bad, but thanks for your time :)
